### ISSUE TITLE
corelist: wrong Digest::SHA version in 5.18.4?

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#15202]
Migrated from [rt.perl.org#127624](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=127624) (status was 'resolved')

Searchable as RT127624$

**Comment by p5pRT** [comment]
### From @mauke

#### Created by @mauke



$ corelist \-v 5\.18\.4 Digest&#8203;::SHA
Digest&#8203;::SHA 5\.84\_01

But
https://metacpan.org/source/RJBS/perl-5.18.4/cpan/Digest-SHA/lib/Digest/SHA.pm
says&#8203;:

$VERSION = '5\.84\_02';

<details><summary>Perl Info</summary>

```

Flags:
    category=library
    severity=low
    module=Module::CoreList

Site configuration information for perl 5.22.1:

Configured by mauke at Tue Dec 29 15:36:05 CET 2015.

Summary of my perl5 (revision 5 version 22 subversion 1) configuration:
   
  Platform:
    osname=linux, osvers=4.2.5-1-arch, archname=i686-linux
    uname='linux simplicio 4.2.5-1-arch #1 smp preempt tue oct 27 08:28:41 cet 2015 i686 gnulinux '
    config_args=''
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=undef, usemultiplicity=undef
    use64bitint=undef, use64bitall=undef, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2',
    cppflags='-fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include'
    ccversion='', gccversion='5.3.0', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234, doublekind=3
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12, longdblkind=3
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -fstack-protector-strong -L/usr/local/lib'
    libpth=/usr/local/lib /usr/lib/gcc/i686-pc-linux-gnu/5.3.0/include-fixed /usr/lib /lib
    libs=-lpthread -lnsl -lnm -lgdbm -ldb -ldl -lm -lcrypt -lutil -lc -lgdbm_compat
    perllibs=-lpthread -lnsl -lnm -ldl -lm -lcrypt -lutil -lc
    libc=libc-2.22.so, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version='2.22'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E'
    cccdlflags='-fPIC', lddlflags='-shared -O2 -L/usr/local/lib -fstack-protector-strong'



@INC for perl 5.22.1:
    /home/mauke/usr/lib/perl5/site_perl/5.22.1/i686-linux
    /home/mauke/usr/lib/perl5/site_perl/5.22.1
    /home/mauke/usr/lib/perl5/5.22.1/i686-linux
    /home/mauke/usr/lib/perl5/5.22.1
    .


Environment for perl 5.22.1:
    HOME=/home/mauke
    LANG=en_US.UTF-8
    LANGUAGE=en_US
    LC_COLLATE=C
    LC_MONETARY=de_DE.UTF-8
    LC_TIME=de_DE.UTF-8
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/mauke/perl5/perlbrew/bin:/home/mauke/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl
    PERLBREW_BASHRC_VERSION=0.73
    PERLBREW_HOME=/home/mauke/.perlbrew
    PERLBREW_ROOT=/home/mauke/perl5/perlbrew
    PERL_BADLANG (unset)
    PERL_UNICODE=SAL
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @jkeenan

On Sat Feb 27 06&#8203;:22&#8203;:44 2016\, mauke\- wrote&#8203;:
> 
> This is a bug report for perl from l\.mai@&#8203;web\.de\,
> generated with the help of perlbug 1\.40 running under perl 5\.22\.1\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please describe your issue here\]
> 
> $ corelist \-v 5\.18\.4 Digest&#8203;::SHA
> Digest&#8203;::SHA 5\.84\_01
> 
> But
> https://metacpan.org/source/RJBS/perl-5.18.4/cpan/Digest-
> SHA/lib/Digest/SHA\.pm
> says&#8203;:
> 
> $VERSION = '5\.84\_02';
> 
> 

Observation confirmed\.

Locally I have available 5\.18\.2 and 5\.18\.4\.  IIRC\, there was a considerable lapse of time between 5\.18\.2 and 5\.18\.3\, but an error in 5\.18\.3 led to the quick issuance of 5\.18\.4\.

The version of Module&#8203;::CoreList shipped with 5\.18\.2 is 3\.03\.  This version correctly has a delta for Digest&#8203;::SHA in 5\.18\.1&#8203;:

\#\#\#\#\#
&nbsp;    5\.018001 => \{
&nbsp;        delta\_from => 5\.018000\,
&nbsp;        changed => \{
&nbsp;            'B'                     => '1\.42\_01'\,
&nbsp;            'Config'                => '5\.018001'\,
&nbsp;            'Digest&#8203;::SHA'           => '5\.84\_01'\,
&nbsp;            'Module&#8203;::CoreList'      => '2\.96'\,
&nbsp;            'Module&#8203;::CoreList&#8203;::TieHashDelta'=> '2\.96'\,
&nbsp;            'Module&#8203;::CoreList&#8203;::Utils'=> '2\.96'\,
&nbsp;        \}\,
&nbsp;        removed => \{
&nbsp;           'VMS&#8203;::Filespec'         => 1\,
&nbsp;        \}
&nbsp;    \}\,
\#\#\#\#\#

The version of Module&#8203;::CoreList shipped with 5\.18\.4 was 3\.13\.  This version lacks any delta for Digest&#8203;::SHA for either 5\.18\.3 or 5\.18\.4&#8203;:

\#\#\#\#\#
&nbsp;    5\.018003 => \{
&nbsp;        delta\_from => 5\.018002\,
&nbsp;        changed => \{
&nbsp;            'Module&#8203;::CoreList'      => '3\.12'\,
&nbsp;            'Module&#8203;::CoreList&#8203;::TieHashDelta'=> '3\.12'\,
&nbsp;            'Module&#8203;::CoreList&#8203;::Utils'=> '3\.12'\,
&nbsp;        \}\,
&nbsp;    \}\,
&nbsp;    5\.018004 => \{
&nbsp;        delta\_from => 5\.018003\,
&nbsp;        changed => \{
&nbsp;            'Module&#8203;::CoreList'      => '3\.13'\,
&nbsp;            'Module&#8203;::CoreList&#8203;::TieHashDelta'=> '3\.13'\,
&nbsp;            'Module&#8203;::CoreList&#8203;::Utils'=> '3\.13'\,
&nbsp;        \}\,
&nbsp;    \}\,
\#\#\#\#\#

This appears to explain why 'corelist \-v 5\.18\.4 Digest&#8203;::SHA' fails to detect Digest&#8203;::SHA version 5\.84\_02\.

Whether we can or should try to fix this at this late date should be discussed\.

Thank you very much\.

\-\- 
James E Keenan \(jkeenan@&#8203;cpan\.org\)

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From zefram@fysh.org

This was fixed in commit b564105ba1218c3a9b4c1f999a8e542b4e9a2d08\,
in perl 5\.23\.9\.

\-zefram

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'resolved'
